File: Dev Priya Gupta/Stack/1047. Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string res="";
        stack<char> stack;
        for(int i=0;i<s.length();i++)
        {
            if(!stack.empty() && stack.top()==s[i])
                stack.pop();
            else
                stack.push(s[i]);
        }
        while(!stack.empty())
        {
            res=stack.top()+res;
            stack.pop();
        }
        return res;
    }
};

Time Complexity : O(n)
Space Complexity : O(n)

File: Riya khandelwal/Stack/1047. Remove All Adjacent Duplicates In String.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> stack = new Stack<>();
        char[] st = s.toCharArray();

        for (int i = 0; i < st.length; i++) {
            if (!stack.isEmpty() && stack.peek() == st[i]) {
                stack.pop();
            } else {
                stack.push(st[i]);
            }
        }

        StringBuilder s1 = new StringBuilder();
        while (!stack.isEmpty()) {
            s1.append(stack.pop());
        }

        return s1.reverse().toString();
    }
}

//T.C = O(n)

//S.C = O(n)
